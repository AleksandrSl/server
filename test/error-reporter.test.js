var errorReporter = require('../error-reporter')
var reporter = require('../reporter')

function errorHelperOut () {
  // use local copy of Jest newline normalization function
  // until Jest doens't apply normalization on comprasion
  return errorReporter.apply({}, arguments).replace(/\r\n|\r/g, '\n')
}

var BaseServer = require('../base-server')

var app = new BaseServer({
  env: 'development',
  pid: 21384,
  nodeId: 'server:H1f8LAyzl',
  subprotocol: '2.5.0',
  supports: '2.x || 1.x'
})
app.listenOptions = { host: '127.0.0.1', port: 1337 }

var originNow = reporter.now
beforeAll(function () {
  reporter.now = function () {
    return new Date((new Date()).getTimezoneOffset() * 60000)
  }
})
afterAll(function () {
  reporter.now = originNow
})

it('handle EACCESS error', function () {
  expect(errorHelperOut({code: 'EACCES'}, app)).toMatchSnapshot()
})

it('handle error in production', function () {
  var http = new BaseServer({
    env: 'production',
    pid: 21384,
    nodeId: 'server:H1f8LAyzl',
    subprotocol: '2.5.0',
    supports: '2.x || 1.x'
  })
  http.listenOptions = { host: '127.0.0.1', port: 1000 }

  expect(errorHelperOut({code: 'EACCES', port: 1000}, http)).toMatchSnapshot()
})

it('handle EADDRINUSE error', function () {
  expect(errorHelperOut({
    code: 'EADDRINUSE',
    port: 1337
  }, app)).toMatchSnapshot()
})

it('handle thorw on undefined error', function () {
  var e = {
    code: 'EAGAIN',
    message: 'resource temporarily unavailable'
  }
  function errorHelperThrow () {
    errorHelperOut(e, app)
  }
  expect(errorHelperThrow).toThrowErrorMatchingSnapshot()
})
