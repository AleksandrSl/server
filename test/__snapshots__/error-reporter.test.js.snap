exports[`test handle EACCESS error 1`] = `
"[1m[31m[40m[7m ERROR [27m[49m[39m[22m [1m[31mYou are not allowed to run server on this port[39m[22m [2mat 1970-01-01 00:00:00[22m
        Try to change user (e.g. root) or use port >= 1024

"
`;

exports[`test handle EADDRINUSE error 1`] = `
"[1m[31m[40m[7m ERROR [27m[49m[39m[22m [1m[31mPort :1337 already in use[39m[22m [2mat 1970-01-01 00:00:00[22m
        Another Logux server or other app already running on this port
        Maybe you didnâ€™t not stop server from other project
        or previous version of this server was not killed.

"
`;

exports[`test handle error in production 1`] = `
" ERROR  You are not allowed to run server on this port at 1970-01-01 00:00:00
        Try to change user (e.g. root) or use port >= 1024

"
`;

exports[`test handle thorw on undefined error 1`] = `"resource temporarily unavailable"`;
